projects/storefrontlib/cms-components/navigation/scroll-to-top/scroll-to-top.component.spec.ts

```diff
@@ -0,0 +1,134 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { Component, DebugElement } from '@angular/core';
+import {
+  WindowRef,
+  CmsScrollToTopComponent,
+  ScrollBehavior,
+  I18nTestingModule,
+} from '@spartacus/core';
+import { CmsComponentData } from '../../../cms-structure/page/model/cms-component-data';
+import { ScrollToTopComponent } from './scroll-to-top.component';
+import { IconTestingModule } from '../../misc/icon/testing/icon-testing.module';
+import { of } from 'rxjs';
+
+@Component({
+  template: `
+    <div style="height: 2000px;">
+      <button class="test"></button>
+      <h1>Test page</h1>
+      <cx-scroll-to-top></cx-scroll-to-top>
+    </div>
+  `,
+})
+class MockComponent {}
+
+class MockWinRef {
+  nativeWindow = window;
+}
+
+const mockData: CmsScrollToTopComponent = {
+  scrollBehavior: ScrollBehavior.SMOOTH,
+  displayThreshold: 100,
+};
+
+const MockCmsComponentData = <CmsComponentData<any>>{
+  data$: of(mockData),
+};
+
+describe('ScrollToTopComponent', () => {
+  let component: MockComponent;
+  let fixture: ComponentFixture<MockComponent>;
+  let winRef: WindowRef;
+  let el: DebugElement;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [IconTestingModule, I18nTestingModule],
+      declarations: [MockComponent, ScrollToTopComponent],
+      providers: [
+        {
+          provide: CmsComponentData,
+          useValue: MockCmsComponentData,
+        },
+        { provide: WindowRef, useClass: MockWinRef },
+      ],
+    }).compileComponents();
+
+    winRef = TestBed.inject(WindowRef);
+    fixture = TestBed.createComponent(MockComponent);
+
+    component = fixture.componentInstance;
+    el = fixture.debugElement;
+  });
+
+  function getKeyboardFocusableElement(): HTMLElement {
```

Comment:
Suggestion: Perhaps add something to the name that will explain more what the function will do (get the top/first focusable element), like `getFirstKeyboardFocusableElement`